MantidQt/CustomInterfaces/src/Reflectometry/QtReflMainView.cpp

```diff
@@ -127,7 +127,16 @@ void QtReflMainView::setTableList(const std::set<std::string> &tables) {
 
     // Map this action to the table name
     m_openMap->setMapping(openTable, QString::fromStdString(*it));
-
+    // we disconnect the signals first so that there are not repeated
+    // connections.
+    // When repeated corrections happen the QMessageBox from openTable()
+    // method in ReflMainViewPresenter will be called multiple times
+    // when 'no' is clicked.
+    disconnect(openTable, SIGNAL(triggered()), m_openMap, SLOT(map()));
+    disconnect(m_openMap, SIGNAL(mapped(QString)), this,
+               SLOT(setModel(QString)));
+    // reconnecting the signals after disconnecting them to ensure
+    // each object has only one of these signals.
```

Comment:
Is this the same as using `Qt::UniqueConnection` when you connect signals? See  [here](http://doc.qt.io/qt-4.8/qt.html#ConnectionType-enum).